### DISCUSSION TITLE
How can I purge all Compare Exchange values

### DISCUSSION THREAD

**Discussion by anemitoff** [ravendb/ravendb#19298] [Q&A] [answered]
I need to perform a maintenance operation where I need to bulk delete my existing compare exchange values and then recreate them with a different key.  The only methods I can find to delete these values is one at a time.

**Comment by ayende** [comment, marked as answer]
There is no other way to delete compare exchange values than by name. What you are doing is the way to go
Note that you can batch those operations using a session, of course
